client/src/use/useMarkerLayer.js

```diff
@@ -6,7 +6,7 @@ export default function useMarkerLayer({ filteredDetections, selectedTrackId })
 
   const markerData = computed(() => {
     if (filteredDetections.value.length === 0) {
-      return null;
+      return [];
```

Comment:
You changed this from a falsey to a truthy value, causing the template `v-if` to change meaning.